### ISSUE TITLE
Ruffle still didn't work in FlashThemes.

### ISSUE THREAD

**Issue by Seven71987** [ruffle-rs/ruffle#17365]
I was going to [FlashThemes](flashthemes.net)[ and when I came to use ruffle to play FlashThemes for testing.
FT Team are aware of Ruffle but I don't think Ruffle is aware of FlashThemes, please respond to this.

**Comment by danielhjacobs** [comment]
Duplicate of #13974. The issue here is a bit complicated. The error mentions `TypeError: Cannot set property isPlaying of #<RufflePlayer> which has only a getter`. That's because Ruffle has a getter called `isPlaying` on its player element, which does not have a setter. However, FlashThemes uses something called ExternalInterface (an API of Flash) to create a function called `isPlaying` on its home page SWF file. When EI creates a function, it typically replaces any identically named Ruffle function. However, it still sees the function signature that says only a getter exists, so when the file then tries to set `isPlaying`, it fails and panics. The only way to fix this, I think, would be for Ruffle to delete any existing player function with an identical name entirely before using EI to create a new one. We need to look into how to do that.

This problem occurred because Ruffle historically added functions that Flash Player did not have to the embed/object elements. We would remove them if we could, but some websites started relying on them and we don't want to break those websites.

**Comment by Seven71987** [comment]
So in a nutshell, it's because FT & Ruffle don't use a same getter and are incompatible, is that right?

**Comment by AggelosZois1** [comment]
Ruffle is still under development and doesn't work for many Flash programs, including FlashThemes.

**Comment by danielhjacobs** [comment]
This particular issue was fixed August 6, 2024. There may still be other FlashThemes issues though, likely open in other issues.

**Comment by AggelosZois1** [comment]
I can use FlashThemes on Waterfox Classic, but Ruffle can still not enable it.

**Comment by danielhjacobs** [comment]
Is the issue:

- #19183: "Unable to create texture object"
- #17206: "Cannot recursively acquire mutex"
- #13248: "Not enough memory left"
- #18983: "Shader blend should not need stencil buffer"

Or is it something else? Those four have been reported on FlashThemes before.

**Comment by danielhjacobs** [comment]
You also may need to use the option to open the files in a new tab, as https://lightspeed.flashthemes.net/ lacks an Access-Control-Allow-Origin header for https://flashthemes.net/

**Comment by AggelosZois1** [comment]
Ok I'll see what I can do.

**Comment by ItsJayPM** [comment]
> You also may need to use the option to open the files in a new tab, as https://lightspeed.flashthemes.net/ lacks an Access-Control-Allow-Origin header for https://flashthemes.net/

@danielhjacobs

Howdy! We've gone ahead and done that for you all.  Unfortunately, support is still a long ways away... Our codebase is incredibly complex, and in particular focuses alot on some features with incomplete implementations in Ruffle... XML, Flex, and shaders in particular.

If there is any way we can work with y'all to help improve support, feel free to contact us!

**Comment by maxhogan092ndchannel-ops** [comment]
I tried using FlashThemes on my school chromebook, and everything's blocked on the chrome web store, and I can't get anything Flash-related, help me
